core/src/main/scala/kafka/server/ConfigAdminManager.scala

```diff
@@ -495,10 +495,7 @@ object ConfigAdminManager {
         case OpType.APPEND => {
           if (!listType(alterConfigOp.configEntry.name, configKeys))
             throw new InvalidRequestException(s"Config value append is not allowed for config key: ${alterConfigOp.configEntry.name}")
-          val oldValueList = Option(configProps.getProperty(alterConfigOp.configEntry.name))
-            .orElse(Option(ConfigDef.convertToString(configKeys(configPropName).defaultValue, ConfigDef.Type.LIST)))
-            .getOrElse("")
-            .split(",").toList
+          val oldValueList = getConfigPropertyAsList(configProps, configKeys, configPropName)
           val newValueList = oldValueList ::: alterConfigOp.configEntry.value.split(",").toList
```

Comment:
Do you think that we should also `trim` the value that is getting appended? If I understand this code correctly, Kafka will store the  white space but a future `APPEND` will trim them.